### PR TITLE
Fixed test runtime calculations.

### PR DESCRIPTION
**Pull request by jposwiata**
Test runtime (defined as current_test_startime -> current_test_endtime) is calculated as close the test as possible to avoid false positives with selftests (where it is expected that the test runs not shorter than given duration). In previous time 'endtime' included framework overhead, while 'starttime' didn't, what "randomly" resulted with CI failures

### CODE REVIEW

File: framework/sandstone.cpp [hunk 1]
```diff
@@ -2120,7 +2118,12 @@ TestResult run_one_test(int *tc, const struct test *test, SandstoneApplication::
     /* First we go to do our -- possibly fractured -- normal run, we'll do retries after */
     for (sApp->current_iteration_count = 0;; ++sApp->current_iteration_count) {
         init_internal(test);
+
+        // calculate starttime->endtime, reduce the overhead to have better test runtime calculations
+        sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration - runtime, &sApp->current_test_starttime);
         state = run_one_test_once(tc, test);
+        runtime += MonotonicTimePoint::clock::now() - sApp->current_test_starttime;
```
Comment (jposwiata) [inline framework/sandstone.cpp:2125]: internal init/cleanup aren't included in the test runtime, but overall runtime includes all the overhead stuff.

Comment (thiagomacieira) [reply framework/sandstone.cpp:2125]: Fair.

File: framework/sandstone.cpp [hunk 2]
```diff
@@ -2170,8 +2173,8 @@ TestResult run_one_test(int *tc, const struct test *test, SandstoneApplication::
             if (sApp->total_retest_count > 0)
                 --sApp->total_retest_count;
             sApp->current_iteration_count = -iterations;
-            sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration);
             init_internal(test);
+            sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration, &sApp->current_test_starttime);
```
Comment (thiagomacieira) [inline framework/sandstone.cpp:2177]: Why are you passing the starttime here, when the previous code didn't?

**Review by thiagomacieira — APPROVED** [review]
Looks ok, so approving, but please answer the question in the comment.

File: framework/sandstone.cpp (hunk 2 above)
Comment (jposwiata) [reply framework/sandstone.cpp:2177]: This allows to set both from exactly same point of time (t.i. end_time := (start_time := now()) + duration). Having it split might provide some unintentional delays.

**Pull request merged**